### PR TITLE
MI-85: Multi package managers & improvements

### PR DESCRIPTION
**Pull request by kai-nguyen-aligent**
This PR is a bit big because I also fix a couple of issues while using this pipeline:

1. Support all 3 main package managers (npm, pnpm & yarn).
2. Update documents as we now not only support Nx monorepo in this pipeline. We also support the [legacy polyrepo] (https://github.com/aligent/nx-serverless-deploy-pipe/pull/8)
3. We used to rely on `ts-node` instead of trans-pile to Javascript.
 - I have removed `ts-node` from `dependencies`.
 - Added `tsx` to `devDependencies` for development.
 - Transpile the code to Javascript before building docker image.
 - We now run the pipeline directly using node. In the future, we also can pack this into a `npm` package and run it directly from Bitbucket pipeline via `npx`.
4. Remove redundant `pipe.yml`. @tvhees I have no idea why this file exist in this repository. Do you recall why it's here & should it be removed or not?
5. We used to only see plain white text even though Bitbucket pipeline support colours. Therefore, I added colour (and a small icon) into our pipe main `console.log()` & preserve colours from sub-commands. The result now look like so:
image

@TheOrangePuff Will need your help on reviewing parts related to docker image building & workflow.

### CODE REVIEW

File: .github/workflows/build.yml
Comment (daniel-vdp) [inline .github/workflows/build.yml]: What does this do? Does it commit the a new package-lock every execution?

File: Dockerfile
Comment (daniel-vdp) [inline Dockerfile]: I'd prefer this default to nothing so we're alerted. I'd hate to have an issue with our docker hub config and have latest override all our images

File: Dockerfile
Comment (daniel-vdp) [inline Dockerfile]: Why are we doing this on so many files? :sweat_smile:

**Review by ryanrixxh — APPROVED** [review]
LGTM but need to run this by a DevOps

File: Dockerfile
Comment (daniel-vdp) [inline Dockerfile]: Imo the build should be done in the docker file rather than outside and copied in via dist

**Review by daniel-vdp — CHANGES_REQUESTED** [review]
Couple comments

File: .github/workflows/build.yml
Comment (kai-nguyen-aligent) [reply .github/workflows/build.yml]: @TheOrangePuff: No. It only clean up `node_modules` folder:
- This command removes packages that are not listed in the dependencies section of your `package.json`.
- It's useful when you want to clean up your `node_modules` folder by removing development dependencies.
- It's often used in deployment scripts or CI/CD pipelines to ensure only production dependencies are included.
- This command doesn't install or update any packages; it only removes unnecessary ones.

File: Dockerfile
Comment (kai-nguyen-aligent) [reply Dockerfile]: No problem, I add this as I get a warning while building the image:
```bash
1 warning found (use docker --debug to expand):
 - InvalidDefaultArgInFrom: Default value for ARG node:${NODE_TAG}-alpine results in empty or invalid base image name (line 2)
```

File: Dockerfile
Comment (kai-nguyen-aligent) [reply Dockerfile]: I'll try to remove the `*.js` part to see if it still work or not. I do this just because it was similar before:
```
RUN chmod a+x ./pipe/*.ts entrypoint.sh
```

**Comment by kai-nguyen-aligent** [comment]
@TheOrangePuff All issue has been addressed. I also update `Dockerfile` to match with recommended approach from https://github.com/nodejs/docker-node/blob/main/docs/BestPractices.md#smaller-images-without-npmyarn.

**Comment by kai-nguyen-aligent** [comment]
@TheOrangePuff Is there anything that block this PR to be merged? I understand that we do not have the final decision on which package manager we should use yet but this PR supports all 3 main one. Considering the following benefit, I think there is no harm of merging this even if we are going to stick with `npm` in the future:
- Support colour in CI
- More robust CI as `ts-node` has been removed. We run the code directly.

**Review by daniel-vdp — APPROVED** [review]
LGTM

**Pull request merged**
